### ISSUE TITLE
Improve autocompletion when adding a new word infront of an existing word

### ISSUE THREAD

**Issue by trevor87** [AnySoftKeyboard/AnySoftKeyboard#1072]
### Steps to reproduce
1. Type "a dog"
2. Position cursor (shown as #): "a #dog"
3. try to write "nice",  so that in the end it will show "a nice #dog"-> While typing "nice" I can't use autocomplete to type the word "nice" because Anysoft Keyboard thinks "dog" belongs to the new word (I get suggestions like ndog, nidog, nicdog...)

### Expected behaviour
Instead when adding a new word before an old one, the old word should be ignored in the autocompletion. So when writing "ni#dog" in the above example the word "nice" should get suggested.

### Actual behaviour
See "steps to reproduce"

**Android OS version:** 5

**Comment by yaomtc** [comment]
I agree, this should be the default behavior. I'd never go to the start of a word to add letters.

**Comment by yaomtc** [comment]
I just tested it. ASK does not give those weird suggestions anymore. However when the cursor is at the start of a word, it doesn't give suggestions at all. A step in the right direction, at least.

For now, you'll have to put the cursor one spot to the left (a# dog) and add another space before you start typing.
